### PR TITLE
HPCC-31530 Remove stale LZ groups

### PR DESCRIPTION
**Pull request by jakesmith**
Remove existing Dali LZ groups before creating groups from the environment or storage planes.
This avoids stale groups being left behind, which in-turn can cause systems to spuriously see them and try to reference nodes that no longer exist, causing stalls and delays to be introduced.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by ghalliday — APPROVED** [review]
@jakesmith looks like a good change.  A couple of suggestions for improving reuse/optimizing it.

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -10525,6 +10525,18 @@ class CInitGroups
         }
         return true;
     }
+    void clearLZGroups()
+    {
+        if (!writeLock)
+            throw makeStringException(0, "CInitGroups::clearLZGroups called in read-only mode");
+        IPropertyTree *root = groupsconnlock.conn->queryRoot();
+        std::vector<Owned<IPropertyTree>> toDelete;
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp:10533]: agreed, I'll change it.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -10525,6 +10525,18 @@ class CInitGroups
...
+        Owned<IPropertyTreeIterator> groups = root->getElements("Group[@kind='dropzone']");
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp:10534]: >are also several that remove the tree within the iterator. Will that work? (E.g. removeEclHiddenStructs)

I think it should work, the iterators link elements, or continue to use ptrees + xpaths, but I wouldn't advise deleting items from an active iterator.

Good idea to enter a dedicated method, but it and refactoring code that uses it, yes in a new PR. I'll open a JIRA.

**Comment by jakesmith** [comment]
@ghalliday - please see new commit. Let me know if you are happy for me to squash it.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**